### PR TITLE
App profiles

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/profiles/EditProfileFragment.java [hunk 1]
```diff
@@ -0,0 +1,831 @@
...
+			}
+		}
+		selectedRoutingProfile = routingProfiles.get(pos);
+		navTypeEt.setText(selectedRoutingProfile.getName());
+		profile.setRoutingProfile(selectedRoutingProfile);
+	}
+
+	void activateMode(ApplicationMode mode) {
+		if (!ApplicationMode.values(app).contains(mode)) {
+			StringBuilder s = new StringBuilder(ApplicationMode.DEFAULT.getStringKey() + ",");
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/profiles/EditProfileFragment.java]: ? serialization ?

File: OsmAnd/src/net/osmand/plus/profiles/EditProfileActivity.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	public static final int DELETE_ID = 1010;
+	public static final String EDIT_PROFILE_FRAGMENT_TAG = "editProfileFragment";
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		getMyApplication().applyTheme(this);
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.single_fragment_layout);
+
+        if (savedInstanceState == null) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/profiles/EditProfileActivity.java]: formatting

File: OsmAnd/src/net/osmand/plus/activities/SettingsBaseActivity.java [hunk 3]
```diff
@@ -48,6 +52,7 @@
 	protected OsmandSettings settings;
 	protected final boolean profileSettings;
 	protected List<ApplicationMode> modes = new ArrayList<ApplicationMode>();
+	protected final ArrayList<BaseProfile> appModes = new ArrayList<>();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/activities/SettingsBaseActivity.java]: ArrayList?

File: OsmAnd/src/net/osmand/plus/profiles/EditProfileFragment.java (hunk 1 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/profiles/EditProfileFragment.java]: fixed

File: OsmAnd/src/net/osmand/plus/activities/SettingsBaseActivity.java (hunk 3 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/activities/SettingsBaseActivity.java]: fixed

File: OsmAnd/src/net/osmand/plus/profiles/EditProfileActivity.java (hunk 2 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/profiles/EditProfileActivity.java]: fixed wherever found

File: OsmAnd-java/src/main/java/net/osmand/router/RoutingConfiguration.java [hunk 4]
```diff
@@ -46,20 +49,27 @@
...
+		private Map<String, GeneralRouter> routers = new LinkedHashMap<>();
+		private Map<String, String> attributes = new LinkedHashMap<>();
+		private HashMap<Long, Location> impassableRoadLocations = new HashMap<>();
 
 		// Example
 //		{
 //			impassableRoadLocations.add(23000069L);
 //		}
 
+		public void updateRouters(Builder update) {
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd-java/src/main/java/net/osmand/router/RoutingConfiguration.java]: artefact. fixed

**Pull request merged**